src/literals.py

```diff
@@ -33,3 +36,42 @@ class Ports:
     "SASL_SSL": Ports(9093, 19093),
     "SSL": Ports(9094, 19094),
 }
+
+AvailableStatuses = Literal[
+    "ACTIVE",
+    "SNAP_NOT_INSTALLED",
+    "ZK_NOT_RELATED",
+    "ZK_NOT_CONNECTED",
+    "SNAP_NOT_RUNNING",
+    "ADDED_STORAGE",
+    "REMOVED_STORAGE",
+    "REMOVED_STORAGE_NO_REPL",
+    "ZK_TLS_MISMATCH",
+    "NO_PEER_RELATION",
+    "ZK_NO_DATA",
+    "NO_BROKER_CREDS",
+    "NO_CERT",
+]
+
+
+STATUS: Dict[AvailableStatuses, StatusBase] = {
+    "ACTIVE": ActiveStatus(),
+    "NO_PEER_RELATION": WaitingStatus("no peer relation yet"),
+    "SNAP_NOT_INSTALLED": BlockedStatus(f"unable to install {SNAP_NAME} snap"),
+    "SNAP_NOT_RUNNING": BlockedStatus("snap service not running"),
+    "ZK_NOT_RELATED": BlockedStatus("missing required zookeeper relation"),
+    "ZK_NOT_CONNECTED": BlockedStatus("unit not connected to zookeeper"),
+    "ZK_TLS_MISMATCH": BlockedStatus("tls must be enabled on both kafka and zookeeper"),
+    "ZK_NO_DATA": WaitingStatus("zookeeper credentials not created yet"),
+    "ADDED_STORAGE": ActiveStatus(
+        "manual partition reassignment may be needed to utilize new storage volumes"
+    ),
+    "REMOVED_STORAGE": BlockedStatus(
+        "manual partition reassignment from replicated brokers recommended due to lost partitions on removed storage volumes"
+    ),
+    "REMOVED_STORAGE_NO_REPL": BlockedStatus(
+        "potential log-data loss due to storage removal without replication"
+    ),
+    "NO_BROKER_CREDS": WaitingStatus("internal broker credentials not yet added"),
+    "NO_CERT": WaitingStatus("unit waiting for signed certificates"),
+}
```

Comment:
**review request:** Especially interested in opinions on interface here. I don't love it and am open to ideas

Reply:
Why not using enums? This really seems a good case for that, such that while typing we can use auto-completion:

```
class Statuses(Enum, StatusBase):
    ACTIVE = ActiveStatus(),
    NO_PEER_RELATION = WaitingStatus("no peer relation yet"),
    SNAP_NOT_INSTALLED = BlockedStatus(f"unable to install {SNAP_NAME} snap"),
```

and in the code:

```
Statuses.ACTIVE.value
```

Reply:
Great idea, much better, will do :)